openedx/core/djangoapps/schedules/README.rst

```diff
@@ -34,29 +34,36 @@ The app introduces a Schedule object to the edX codebase. Learners receive
 Schedules when they enroll in self-paced courses. The app uses Schedules to
 determine which learners to message.
 
+In the future, the Schedules app may be extended or changed to support more
+complicated communication patterns. For example, something that takes into
+account the user's progress through the course.
+
 Glossary
 --------
 
--  **Schedule**: Stores the day a learner enrolls in a course and the learner’s
-“upgrade deadline”. Used by the app at runtime to determine which learners to
-message.
+-  **Schedule**: Stores the day a learner enrolls in a course and the
+   learner's "upgrade deadline". This information allows us to personalize
+   learners' experiences in self-paced courses so that events happen relative to
+   the learner's schedule, not the course's schedule.
 
 -  **Schedule Experience**: Defines the set of emails that a learner
    receives. The two Schedule Experiences are:
 
    -  “Recurring Nudge and Upgrade Reminder”
    -  “Course Updates”
 
--  **Upgrade Deadline**: The date before which a learner can purchase a
-   verified certificate. By default, a Schedule imposes a “soft” upgrade
-   deadline 21 days from when a learner enrolled in a course. A course
-   also imposes a “hard” upgrade deadline. A Schedule uses whichever
-   date is earlier.
+-  **Upgrade Deadline**: The date before which a learner is encouraged to
+   purchase a verified certificate. By default, a Schedule imposes a "soft"
+   upgrade deadline (meaning, a suggested, but not final, date) 21 days from
+   when a learner enrolled in a course. A self-paced course imposes a "hard"
+   upgrade deadline that is the course-wide expiration date for upgrading on the
+   course. A Schedule uses whichever date is earlier.
 
--  **Course Update**: We refer to “Weekly Course Highlight Messages” as “Course
-   Updates” in the code. In contexts outside of this app, Course Updates refer to
-   bulk-emails sent by course instructors. We plan on removing this term from the
-   code to avoid confusion.
+-  **Course Update**: We refer to "Weekly Course Highlight Messages" as "Course
+   Updates" in the code. In contexts outside of this app, Course Updates refer
+   to bulk-emails manually sent out by course instructors and for the blocks of
```

Comment:
We should break this into two sentences. "Course Updates can also refer to the blocks of text that course instructors add to the top of their course outlines."